tests/test_tokenizer.py

```diff
@@ -45,3 +56,60 @@ def test_get_lines():
         if t.getnext().type == ENDMARKER:
             break
     assert t.get_lines([1, 2, 3]) == ["1\n", "2\n", "3"]
+
+
+def test_dedent():
+    source = io.StringIO(
+        """
+def test(num):
+    a = pmatch num:
+        1: "One"
+        2: "Two"
+        3: "Three"
+        _: "Number not between 1 and 3"
+
+    return a
+"""
+    )
+    t = Tokenizer(generate_tokens(source.readline))
+    expected = [
+        NAME,
+        NAME,
+        OP,
+        NAME,
+        OP,
+        OP,
+        NEWLINE,
+        INDENT,
+        NAME,
+        OP,
+        NAME,
+        NAME,
+        OP,
+        NEWLINE,
+        INDENT,
+        NUMBER,
+        OP,
+        STRING,
+        NEWLINE,
+        NUMBER,
+        OP,
+        STRING,
+        NEWLINE,
+        NUMBER,
+        OP,
+        STRING,
+        NEWLINE,
+        NAME,
+        OP,
+        STRING,
+        NEWLINE,
+        DEDENT,
```

Comment:
I would like the tokenizer to output DEDENT + NEWLINE here.

The would allow the parser I'm working on to parse

```
pmatch-statement NEWLINE return-statement
```

as two statements separated by a newline. The current tokenizer behavior makes it awkward to parse.